app/models/user_coffee.rb

```diff
@@ -15,9 +15,12 @@ def name
     coffee.name
   end
 
-  private
+  def coffee_details
+    "#{coffee.roaster} | #{coffee.name} | #{coffee.roasting_date} | ~ #{remaining_quantity} gr left"
+    # "#{coffee.roaster} | #{coffee.name} | #{coffee.roasting_date}"
+  end
 
-  def set_weight_left
-    self.weight_left = self.attributes['bag_weight']
+  def remaining_quantity
+    bag_weight - extractions.sum(:weight_in)
```

Comment:
we have a weight_left column on user_coffee which would correspond to this remaining_quantity
Maybe it isn't necessary...